flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServicesConfiguration.java

```diff
@@ -104,6 +113,11 @@ public TaskManagerServicesConfiguration(
 			boolean preAllocateMemory,
 			float memoryFraction,
 			int pageSize,
+			MemorySize taskHeapMemorySize,
+			MemorySize taskOffHeapMemorySize,
+			MemorySize shffuleMemorySize,
+			MemorySize onHeapManagedMemorySize,
+			MemorySize offHeapManagedMemorySize,
```

Comment:
Maybe we could directly pass `taskExecutorResourceSpec ` to `TaskManagerServicesConfiguration` or add a `Builder` for it? Since the constructor takes too many parameters which easily become unwieldy.